src/Icicle/Avalanche/Prim/Eval.hs

```diff
@@ -111,7 +111,7 @@ evalPrim p vs
       -> primError
 
 
-     PrimUpdate (PrimUpdateArrayPut t)
+     PrimArray (PrimArrayPutMutable t)
```

Comment:
this should really be mutating but it would require the evaluator to take a heap. this is good for now